api/edge_api/identities/edge_request_forwarder.py

```diff
@@ -4,84 +4,99 @@
 from core.constants import FLAGSMITH_SIGNATURE_HEADER
 from core.signing import sign_payload
 from django.conf import settings
-from rest_framework.request import Request
 
 from environments.dynamodb.migrator import IdentityMigrator
-from util.util import postpone
+from task_processor.decorators import register_task_handler
 
 
 def _should_forward(project_id: int) -> bool:
     migrator = IdentityMigrator(project_id)
     return bool(migrator.is_migration_done)
 
 
-@postpone
-def forward_identity_request(request, project_id: int):
-    forward_identity_request_sync(request, project_id)
-
-
-def forward_identity_request_sync(request, project_id: int):
+@register_task_handler()
+def forward_identity_request(
+    request_method: str,
+    headers: dict,
+    project_id: int,
+    query_params: dict = None,
+    request_data: dict = None,
+):
     if not _should_forward(project_id):
         return
 
     url = settings.EDGE_API_URL + "identities/"
-    forwarding_method = (
-        _forward_identity_post_request
-        if request.method == "POST"
-        else _forward_identity_get_request
-    )
-    forwarding_method(request, url)
+    if request_method == "POST":
+        _forward_identity_post_request(request_method, headers, url, request_data)
+        return
+    _forward_identity_get_request(request_method, headers, url, query_params)
 
 
-def _forward_identity_get_request(request: Request, url: str):
+def _forward_identity_get_request(
+    request_method: str, headers: dict, url: str, query_params: dict
```

Comment:
It feels a bit odd to include the `request_method` argument in a function in which we know the request method is GET, right? Couldn't we get the headers in the `forward_identity_request` function and provide them to the relevant function directly? 

```python
    headers = _get_headers(request_method, headers)
    if request_method == "POST":
        _forward_identity_post_request(headers, url, request_data)
        return
    _forward_identity_get_request(headers, url, query_params)
```